articles/external_qos_file.md

```diff
@@ -0,0 +1,658 @@
+
+```xml
+<qos_profiles>
+    <default>
+        <publisher topic_name="/my_ns/nested_ns/asd">
+            ...
+        </publisher>
+    </default>
+    <node name="my_node" namespace="/my_ns/nested_ns">
+        <publisher topic_name="asd">
+            ...  <!--QoS settings here will override the ones in `default`. The settings specified in `default` will be used as a base.-->
+        </publisher>
+    </node>
+</qos_profiles>
+```
+
+```yaml
+/**:
+    ros__qos_profiles:
+        publisher:
+            topic_name: /my_ns/nested_ns/asd
+            qos:
+                ...
+        subscription:
+            topic_name: /another_ns/asd
+            qos:
+                ...
+/my_ns/nested_ns/my_node:
+    ros__qos_profiles:
+        publisher:
+            topic_name: asd
+            qos:
+                ...
+        subscription:
+            topic_name: /another_ns/asd
+            qos:
+                ...
+```
+
+## Named QoS profiles
+
+The file can also include named QoS profiles, that can be referenced within the same file:
+
+```xml
+<qos_profiles>
+    <profiles>
+        <qos name="reliable_depth_100">
+            <reliability>reliable</reliability>
+            <history_depth>100</history_depth>
+            <history>keep_last</history>
+        </qos>
+    </profiles>
+    <node name="my_node" namespace="/my_ns/nested_ns">
+        <publisher topic_name="asd">
+            <qos base="reliable_depth_100"/>  <!--Uses exactly reliable_depth_100 profile-->
+        </publisher>
+        <publisher topic_name="bsd">
+            <qos base="reliable_depth_100">  <!--Uses reliable_depth_100 profile as a base, and overrides the durability policy-->
+                <lifespan>10s</lifespan>
+            </qos>
+        </publisher>
+    </node>
+</qos_profiles>
+```
+
+```yaml
+/**:
+    ros__qos_profiles:
+        profiles:
+            reliable_depth_100:
+                reliability: reliable
+                history_depth: 100
+                history: keep_last
+/my_ns/nested_ns/my_node:
+    ros__qos_profiles:
+        publisher:
+            topic_name: asd
+            qos:
+                base: reliable_depth_100
+        publisher:
+            topic_name: bsd
+            qos:
+                base: reliable_depth_100
+                lifespan: 10s
+```
+
+This mechanism will help ensuring matching QoS profile, where that's needed in the system.
+
+Having some implicitly defined named profiles can become handy to easily write profiles.
+The following could be implicitly defined:
+
+- ros_default: Equivalent to `rmw_qos_profile_default`.
+- ros_sensor_data: Equivalent to `rmw_qos_profile_sensor_data`.
+- ros_service_default: Equivalent to `rmw_qos_profile_services_default` (as of ROS 2 Foxy, the same as `ros_default`).
+- ros_system_default: Equivalent to `rmw_qos_profile_system_default`.
+
+See [qos_profiles.h](https://github.com/ros2/rmw/blob/master/rmw/include/rmw/qos_profiles.h).
```

Comment:
> It might be better to use a permalink in case the filename or location changes in the future.

```suggestion
See [qos_profiles.h](https://github.com/ros2/rmw/blob/0176872b10ff502ae34fa3eef5e900145f18b0b3/rmw/include/rmw/qos_profiles.h).
```